x-pack/plugin/ql/src/main/java/org/elasticsearch/xpack/ql/index/IndexResolver.java

```diff
@@ -676,23 +702,38 @@ private static FieldCapabilitiesRequest createFieldCapsRequest(
         IndicesOptions indicesOptions,
         Map<String, Object> runtimeMappings
     ) {
-        return new FieldCapabilitiesRequest().indices(Strings.commaDelimitedListToStringArray(index))
+        return createFieldCapsRequest(index, fieldNames, indicesOptions, runtimeMappings, null);
+    }
+
+    private static FieldCapabilitiesRequest createFieldCapsRequest(
+        String index,
+        Set<String> fieldNames,
+        IndicesOptions indicesOptions,
+        Map<String, Object> runtimeMappings,
+        String projectRouting
+    ) {
+        FieldCapabilitiesRequest result = new FieldCapabilitiesRequest().indices(Strings.commaDelimitedListToStringArray(index))
             .fields(fieldNames.toArray(String[]::new))
             .includeUnmapped(true)
             .runtimeFields(runtimeMappings)
             // lenient because we throw our own errors looking at the response e.g. if something was not resolved
             // also because this way security doesn't throw authorization exceptions but rather honors ignore_unavailable
             .indicesOptions(indicesOptions);
+        if (projectRouting != null) {
+            result.projectRouting(projectRouting);
+        }
```

Comment:
It is okay to set null project routing (meaning no routing).
It might be a bit simpler, but this way it is obvious that it is optional and might not be present in some requests.